### PR TITLE
Clearing accounts cache when switching networks.

### LINKED ISSUE #25: Clear caches on network change

### PR DESCRIPTION
**Pull request by tomusdrw**
Correctly handling disconnected status.
Closes #25

Since lookup is using a pre-defined address anyway, I don't clean caches for e-mail resolutions.

### CODE REVIEW

File: background/transport.js [hunk 1]
```diff
@@ -60,11 +60,24 @@ let accountsCache = {};
 function newTransport (token) {
   const transport = new Ws(`ws://${UI}`, token, true);
   transport.on('open', () => {
+    let oldOrigins = Object.keys(accountsCache);
```
Comment (ngotchac) [inline background/transport.js]: Use `const` ?

File: web3/content_script.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
+/* global chrome */
```
Comment (ngotchac) [inline web3/content_script.js:17]: Really not sure, but shouldn't the Parity GPL be included for this repo ?

File: web3/index.js [hunk 3]
```diff
@@ -1,158 +1,76 @@
...
-    if (!ev.data.type || ev.data.type !== EV_SIGNER_BAR) {
-      return;
-    }
-    if (ev.data.opened) {
-      iframe.classList.add(styles.iframe__open);
-    } else {
-      iframe.classList.remove(styles.iframe__open);
-    }
+function getBackgroundSeed (callback) {
+  chrome.storage.local.get('backgroundSeed', res => {
```
Comment (ngotchac) [inline web3/index.js]: Why not having Promises here ?

**Comment by ngotchac** [comment]
Looks good. Unrelated to this PR, but I think the reconnection logic needs some love (look at the JS console when disconnected).

**Comment by tomusdrw** [comment]
If you mean this:
screenshot from 2017-01-31 15-07-36

Everything here is coming from ParityBar, so it should be fixed in Parity repo (as the code is just fetched from the client)

**Comment by ngotchac** [comment]
I meant that :
image

I think it's only when a dapp is open, and then the connection is closed (eg. on Maker Market). Not a big issue though

**Review by ngotchac — APPROVED** [review]

**Comment by tomusdrw** [comment]
Hmm that's weird. These are logged in background script - it should be visible only if you inspect background script, does it display for you in DevTools opened for a single website?

**Comment by ngotchac** [comment]
Oh yes, that's definitely in the Background script. Just thought that we might want to take a look at why these calls are made even if we are disconnected, and why 50+ `Transport uninitialized` error messages are printed in less than 200ms. I haven't looked at the code yet, so it might just be extra logging, which is fine. I just want to make sure that we don't make too much unnecessary requests.

**Comment by tomusdrw** [comment]
It's because the page you have open is trying to make those calls, but I agree that extra logging and limitting errors would be useful.

**Pull request merged**